### PR TITLE
add connection for inverse proxy

### PR DESCRIPTION
**Pull request by IronPan**
With this change, user can access the cluster through python with single command
   > client = kfp.client(host=[inverse-proxy-url])   
   > client.get_run() ...

/assign @Ark-kun

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1993)

### CODE REVIEW

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -96,12 +98,14 @@ def __init__(self, host=None, client_id=None, namespace='kubeflow'):
 
   def _load_config(self, host, client_id, namespace):
     config = kfp_server_api.configuration.Configuration()
-    host = host or os.environ.get(KF_PIPELINES_ENDPOINT_ENV)
```
Comment (Ark-kun) [inline sdk/python/kfp/_client.py]: Is this env variable no longer used?

File: sdk/python/kfp/_client.py [hunk 2]
```diff
@@ -136,6 +140,14 @@ def _load_config(self, host, client_id, namespace):
       config.host = os.path.join(config.host, Client.KUBE_PROXY_PATH.format(namespace))
     return config
 
+  def _is_iap_host(self, host, client_id):
+    if host and client_id:
+      return re.match(r'\S+.endpoints.\S+.cloud.goog', host)
+    return False
+
+  def _is_inverse_proxy_host(self, host):
+    return re.match(r'\S+dot-datalab-vm\S+.googleusercontent.com', host)
```
Comment (Ark-kun) [inline sdk/python/kfp/_client.py]: Is it always `datalab-vm`?

File: sdk/python/kfp/_client.py (hunk 1 above)
Comment (IronPan) [reply sdk/python/kfp/_client.py]: Could you help cross check the caller. the original code looks a bit funky to me.

File: sdk/python/kfp/_client.py (hunk 2 above)
Comment (IronPan) [reply sdk/python/kfp/_client.py]: yes. and if we change later we can match against an array.

File: sdk/python/kfp/_client.py (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/kfp/_client.py]: There was a person who wanted to set entrypoints using env variables (he was probably using shell scripts for automation.).

Comment (IronPan) [reply sdk/python/kfp/_client.py]: that is fine. i pass in self._uihost in caller. should be equavalent.

Comment (IronPan) [reply sdk/python/kfp/_client.py]: the line here is contradict to how we get host in line 90 so can potentially be buggy.

Comment (Ark-kun) [reply sdk/python/kfp/_client.py]: `uihost` is read from `KF_PIPELINES_UI_ENDPOINT_ENV` while the host is read from `KF_PIPELINES_ENDPOINT_ENV`

Comment (Ark-kun) [reply sdk/python/kfp/_client.py]: Maybe you just move the `host = host or os.environ.get(KF_PIPELINES_ENDPOINT_ENV)` up into `__init__`?
```python
host = host or os.environ.get(KF_PIPELINES_ENDPOINT_ENV)
self._uihost = os.environ.get(KF_PIPELINES_UI_ENDPOINT_ENV, host)
config = self._load_config(host, client_id, namespace)
```

Comment (IronPan) [reply sdk/python/kfp/_client.py]: done

**Pull request merged**